src/main/kotlin/cloud/luxor/lbwl/flash/MapVoting.kt

```diff
@@ -70,22 +85,27 @@ class MapVoting(private val maps: List<MapConfig>) : Listener {
     fun determineMap(): MapConfig {
         // choose random map if no-one has voted
         if (this.votes.values.sum() == 0) return this.maps.shuffled()[0]
-    
+
         // find the map with the highest votes
-        val name = this.votes.toList().maxBy { (_, value) -> value }!!
-        return this.maps.find { config -> name.first == config.name }!!
+        val name = this.votes.toList().maxBy { (_, value) -> value }
+        return this.maps.find { config -> name.first.contains(config.name) }!!
     }
 
     @EventHandler
     private fun onInventoryClick(event: InventoryClickEvent) {
         if (event.clickedInventory == null) return
-        if (event.clickedInventory.name != "Maps") return
+        //if((event.clickedInventory?.holder as Container).name != "Text")
         if (event.currentItem == null) return
-        if (event.currentItem.itemMeta == null) return
+        if (event.currentItem!!.itemMeta == null) return
 
         event.isCancelled = true
-        val name = ChatColor.stripColor(event.currentItem.itemMeta.displayName)
-        this.vote(event.whoClicked as Player, name)
+
+        //ugly af but I really don't know how to access the displayName as a String
+        val mapName = event.currentItem?.itemMeta?.displayName()
```

Comment:
Maybe try to cast it into a `TextComponent` and then get the plain string by calling `TextComponent#content()`

Reply:
that solved the problem see [https://github.com/Freggy/lbwl-flash/pull/5/commits/7b0fbea056ff503db4ee7e64c5379881ff1eb863](https://github.com/Freggy/lbwl-flash/pull/5/commits/7b0fbea056ff503db4ee7e64c5379881ff1eb863)